manageiq-operator/pkg/helpers/miq-components/postgresql.go

```diff
@@ -40,80 +44,94 @@ func postgresqlSecretName(cr *miqv1alpha1.ManageIQ) string {
 	return secretName
 }
 
-func NewPostgresqlConfigsConfigMap(cr *miqv1alpha1.ManageIQ) *corev1.ConfigMap {
+func NewPostgresqlConfigsConfigMap(cr *miqv1alpha1.ManageIQ) (*corev1.ConfigMap, controllerutil.MutateFn) {
 	labels := map[string]string{
 		"app": cr.Spec.AppName,
 	}
-	return &corev1.ConfigMap{
+
+	data := map[string]string{
+		"01_miq_overrides.conf": postgresqlOverrideConf(),
+	}
+
+	configMap := &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "postgresql-configs",
+			Name:      "miq-pg-configs",
 			Namespace: cr.ObjectMeta.Namespace,
-			Labels:    labels,
-		},
-		Data: map[string]string{
-			"01_miq_overrides.conf": postgresqlOverrideConf(),
 		},
 	}
+
+	mutateFn := configMapMutateFn(configMap, labels, data)
+
+	return configMap, mutateFn
 }
 
-func NewPostgresqlPVC(cr *miqv1alpha1.ManageIQ) *corev1.PersistentVolumeClaim {
+func NewPostgresqlPVC(cr *miqv1alpha1.ManageIQ) (*corev1.PersistentVolumeClaim, controllerutil.MutateFn) {
 	labels := map[string]string{
 		"app": cr.Spec.AppName,
 	}
+
 	storageReq, _ := resource.ParseQuantity(cr.Spec.DatabaseVolumeCapacity)
+
+	resources := corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			"storage": storageReq,
+		},
+	}
+
+	accessModes := []corev1.PersistentVolumeAccessMode{
+		"ReadWriteOnce",
+	}
+
 	pvc := &corev1.PersistentVolumeClaim{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "postgresql",
+			Name:      "miq-pgdb-volume",
 			Namespace: cr.ObjectMeta.Namespace,
-			Labels:    labels,
-		},
-		Spec: corev1.PersistentVolumeClaimSpec{
-			AccessModes: []corev1.PersistentVolumeAccessMode{
-				"ReadWriteOnce",
-			},
-			Resources: corev1.ResourceRequirements{
-				Requests: corev1.ResourceList{
-					"storage": storageReq,
-				},
-			},
 		},
 	}
 
 	if cr.Spec.StorageClassName != "" {
 		pvc.Spec.StorageClassName = &cr.Spec.StorageClassName
 	}
 
-	return pvc
+	mutateFn := pvcMutateFn(pvc, labels, accessModes, resources)
+
+	return pvc, mutateFn
 }
 
-func NewPostgresqlService(cr *miqv1alpha1.ManageIQ) *corev1.Service {
+func NewPostgresqlService(cr *miqv1alpha1.ManageIQ) (*corev1.Service, controllerutil.MutateFn) {
 	labels := map[string]string{
 		"app": cr.Spec.AppName,
 	}
+
 	selector := map[string]string{
 		"name": "postgresql",
 	}
-	var port int32 = 5432
-	return &corev1.Service{
+
+	ports := []corev1.ServicePort{
+		corev1.ServicePort{
+			Name: "postgresql",
+			Port: 5432,
+		},
+	}
+
+	service := &corev1.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "postgresql",
 			Namespace: cr.ObjectMeta.Namespace,
-			Labels:    labels,
-		},
-		Spec: corev1.ServiceSpec{
-			Ports: []corev1.ServicePort{
-				corev1.ServicePort{
-					Name: "postgresql",
-					Port: port,
-				},
-			},
-			Selector: selector,
 		},
 	}
+
+	mutateFn := serviceMutateFn(service, labels, ports, selector)
```

Comment:
I think you can just inline this into the return value if the function doesn't return an error, no need for an extra variable.